components/scream/src/share/grid/remap/horizontal_remap_utility.hpp

```diff
@@ -0,0 +1,564 @@
+  Int get_num_of_segs() {return map_segments.size();}
+  Int get_num_of_dofs() {return m_dofs_gids.size();}
+/*---------------------------------------------*/
+  RemapSegment get_segment(const gid_type dof) {
+    return get_segment_impl(dof);
+  }
+/*---------------------------------------------*/
+  void set_dofs_gids(const view_1d<gid_type>& dofs_gids, const gid_type min_dof) {
+    EKAT_REQUIRE(dofs_gids.size()>0);
+    m_dofs_gids = view_1d<gid_type>(dofs_gids);
+    m_num_dofs = m_dofs_gids.extent(0);
+    Kokkos::parallel_for("", m_num_dofs, KOKKOS_LAMBDA (const int& ii) {
+      m_dofs_gids(ii) = dofs_gids(ii)-min_dof;
+    });
+  }
+/*---------------------------------------------*/
+  std::vector<gid_type> get_unique_dofs() { return unique_dofs; };
+/*---------------------------------------------*/
+  void add_segment(RemapSegment& segment) {
+    add_segment_impl(segment);
+  }; 
+/*---------------------------------------------*/
+  // Useful for debugging, print all the GSMap info.
+  void print_GSMap() {
+    printf("\n=============================================\n");
+    printf("Printing GSMap information:\n");
+    for (int ii=0; ii<map_segments.size(); ii++) {
+      auto& seg = map_segments[ii];
+      seg.print_seg();
+    } 
+    printf(" Unique dofs info\n");
+    for (int ii=0; ii<unique_dofs.size(); ii++) {
+      printf("%10d: %10d\n",ii, unique_dofs[ii]);
+    }
+    printf(" dofs_gids\n");
+    auto dofs_gids_h = Kokkos::create_mirror_view(m_dofs_gids);
+    Kokkos::deep_copy(dofs_gids_h,m_dofs_gids);
+    for (int ii=0; ii<dofs_gids_h.extent(0); ii++) {
+      printf("%10d: %10d\n",ii,dofs_gids_h(ii));
+    } 
+    printf("\n=============================================\n");
+  }
+/*---------------------------------------------*/
+  void check() {
+    std::vector<bool> found(m_dofs_gids.size(),false);
+    auto dofs_gids_h = Kokkos::create_mirror_view(m_dofs_gids);
+    Kokkos::deep_copy(dofs_gids_h,m_dofs_gids);
+    for (int ii=0; ii<map_segments.size(); ii++) {
+      auto& seg = map_segments[ii];
+      auto seg_check = seg.check();
+      EKAT_REQUIRE_MSG(seg_check,"Error in GSMap " + map_name + " - problem with a remapping segment for dof = " + std::to_string(seg.m_dof) + ".");
+      for (int jj=0; jj<dofs_gids_h.size(); jj++) {
+        if (dofs_gids_h(jj) == seg.m_dof) {
+          found[jj] = true;
+          break;
+        }
+      }
+    }
+    bool pass = std::find(found.begin(),found.end(),false) == found.end();
+    if (!pass) {
+      printf("Error: GSMap.check() for map %s - Target column(s) are missing a remap segment:\n",map_name.c_str());
+      for (int ii=0;ii<m_dofs_gids.size();ii++) {
+        if (!found[ii]) {
+          printf("No segment found for DOF = %d\n",dofs_gids_h(ii));
+        }
+      }
+    }
+    EKAT_REQUIRE(pass);
+  };
+/*---------------------------------------------*/
+  // Useful if needing to grab source data from somewhere, like a data file.
```

Comment:
It might be helpful to document exactly what this function does (in terms of the members of the `GSMap`, with preconditions and postconditions, etc), since it takes no arguments.